odoo/cli/obfuscate.py

```diff
@@ -0,0 +1,247 @@
+            self.cr.execute(query, {"pwd": pwd})
+            if with_commit:
+                self.commit()
+                self.begin()
+
+    def confirm_not_secure(self):
+        _logger.info("The obfuscate method is not considered as safe to transfer anonymous datas to a third party.")
+        conf_y = input(f"This will alter data in the database {self.dbname} and can lead to a data loss. Would you like to proceed [y/N]? ")
+        if conf_y.upper() != 'Y':
+            self.rollback()
+            sys.exit(0)
+        conf_db = input(f"Please type your database name ({self.dbname}) in UPPERCASE to confirm you understand this operation is not considered secure : ")
+        if self.dbname.upper() != conf_db:
+            self.rollback()
+            sys.exit(0)
+        return True
+
+    def run(self, cmdargs):
+        parser = odoo.tools.config.parser
+        group = optparse.OptionGroup(parser, "Populate Configuration")
+        group.add_option('--pwd', dest="pwd", default=False, help="Cypher password")
+        group.add_option('--fields', dest="fields", default=False, help="List of table.columns to obfuscate/unobfuscate: table1.column1,table2.column1,table2.column2")
+        group.add_option('--exclude', dest="exclude", default=False, help="List of table.columns to exclude from obfuscate/unobfuscate: table1.column1,table2.column1,table2.column2")
+        group.add_option('--file', dest="file", default=False, help="File containing the list of table.columns to obfuscate/unobfuscate")
+        group.add_option('--unobfuscate', action='store_true', default=False)
+        group.add_option('--allfields', action='store_true', default=False, help="Used in unobfuscate mode, try to unobfuscate all fields. Cannot be used in obfuscate mode. Slower than specifying fields.")
+        group.add_option('--vacuum', action='store_true', default=False, help="Vacuum database after unobfuscating")
+        group.add_option('--pertablecommit', action='store_true', default=False, help="Commit after each table instead of a big transaction")
+
+        parser.add_option_group(group)
+        if not cmdargs:
+            sys.exit(parser.print_help())
+
+        try:
+            opt = odoo.tools.config.parse_config(cmdargs)
+            if not opt.pwd:
+                _logger.error("--pwd is required")
+                sys.exit("ERROR: --pwd is required")
+            if opt.allfields and not opt.unobfuscate:
+                _logger.error("--allfields can only be used in unobfuscate mode")
+                sys.exit("ERROR: --allfields can only be used in unobfuscate mode")
+            self.dbname = odoo.tools.config['db_name']
+            self.registry = odoo.registry(self.dbname)
+            with self.registry.cursor() as cr:
+                self.cr = cr
+                self.begin()
+                if self.check_pwd(opt.pwd):
+                    fields = [
```

Comment:
Can we delegate this list to a method? :pray:

Reply:
> Can we delegate this list to a method? 🙏

An overloadable method. (if the method is in ``obfuscate.py``  it will not be possible to overload it in community module).

Reply:
If I'm not mistaken this is a Command class and you could reuse it in your own command. Am I wrong?

Reply:
@legalsylvain I understand. I was trying to find a way that would not made them reply w/ "we don't touch models"  or something like that :wink: 
But I agree... it could be that if a model has `_obfuscate_fields_get` method you get the list from there.